src/Modules/CalcOffence.lua

```diff
@@ -3473,27 +3473,72 @@ function calcs.offence(env, actor, activeSkill)
 			skillFlags.chill = true
-			output.ChillEffectMod = skillModList:Sum("INC", cfg, "EnemyChillEffect")
-			output.ChillDurationMod = 1 + skillModList:Sum("INC", cfg, "EnemyChillDuration") / 100
-			output.ChillSourceEffect = m_min(skillModList:Override(nil, "ChillMax") or ailmentData.Chill.max, m_floor(ailmentData.Chill.default * (1 + output.ChillEffectMod / 100)))
+			local chillDurationBase = skillModList:Sum("BASE", cfg, "EnemyChillDuration") or ailmentData.Chill.duration
+			local chillEffectBase = skillModList:Sum("BASE", cfg, "EnemyChillEffect") or ailmentData.Chill.default
+			local incDur = skillModList:Sum("INC", cfg, "EnemyChillDuration") + enemyDB:Sum("INC", nil, "SelfChillDuration")
+			local moreDur = skillModList:More(cfg, "EnemyChillDuration") * enemyDB:More(nil, "SelfChillDuration")
+			output.ChillDurationMod = (1 + incDur / 100) * moreDur * debuffDurationMult
+			output.ChillDuration = chillDurationBase * output.ChillDurationMod
+			skillFlags.chillHasDuration = output.ChillDuration > 0
+			output.ChillEffectMod = calcLib.mod(skillModList, cfg, "EnemyChillEffect")
+			output.ChillSourceEffect = m_min(skillModList:Override(nil, "ChillMax") or ailmentData.Chill.max, m_floor(chillEffectBase * output.ChillEffectMod))
 			if breakdown then
 				breakdown.DotChill = { }
 				breakdown.multiChain(breakdown.DotChill, {
 					label = s_format("Effect of Chill: ^8(capped at %d%%)", skillModList:Override(nil, "ChillMax") or ailmentData.Chill.max),
-					base = s_format("%d%% ^8(base)", ailmentData.Chill.default),
-					{ "%.2f ^8(increased effect of chill)", 1 + output.ChillEffectMod / 100},
+					base = s_format("%d%% ^8(base)", chillEffectBase),
+					{ "%.2f ^8(increased effect of chill)", output.ChillEffectMod},
 					total = s_format("= %.0f%%", output.ChillSourceEffect)
 				})
+				if output.ChillDuration ~= chillDurationBase then
+					breakdown.ChillDuration = { }
+					t_insert(breakdown.ChillDuration, s_format("%.2fs ^8(base duration)", chillDurationBase))
+					if incDur ~= 0 then
+						t_insert(breakdown.ChillDuration, s_format("x %.2f ^8(increased/reduced duration)", 1 + incDur / 100))
+					end
+					if moreDur ~= 1 then
+						t_insert(breakdown.ChillDuration, s_format("x %.2f ^8(more/less duration)", moreDur))
+					end
+					if debuffDurationMult ~= 1 then
+						t_insert(breakdown.ChillDuration, s_format("/ %.2f ^8(debuff expires slower/faster)", 1 / debuffDurationMult))
+					end
+					t_insert(breakdown.ChillDuration, s_format("= %.2fs", output.ChillDuration))
+				end
 			end
 		end
-		if (output.FreezeChanceOnHit + output.FreezeChanceOnCrit) > 0 then
-			if globalBreakdown then
-				globalBreakdown.FreezeDurationMod = {
-					s_format("Ailment mode: %s ^8(can be changed in the Configuration tab)", igniteMode == "CRIT" and "Crits Only" or "Average Damage")
-				}
+		if skillModList:Flag(nil, "Condition:ArcticArmourCanFreeze") then
```

Comment:
Any way we can move these calculations to a preCalc function in ArticArmour specifically?  This is a lot of calculation being added for just one skill.